sops/__init__.py

```diff
@@ -620,6 +632,7 @@ def walk_and_decrypt(branch, key, aad=b'', stash=None, digest=None,
     for k, v in branch.items():
         if k == 'sops' and isRoot:
             continue    # everything under the `sops` key stays in clear
+        v_unencrypted = unencrypted or k.endswith(unencrypted_suffix)
```

Comment:
what about `must_decrypt = not unencrypted and not k.endswith(UNENCRYPTED_SUFFIX)` ?
and then only:
```python
if must_decrypt:
    decrypt(....)
```